module/validation/seal_validator.go

```diff
@@ -0,0 +1,272 @@
+	headers          storage.Headers
+	payloads         storage.Payloads
+	skipSealValidity bool
+}
+
+func NewSealValidator(state protocol.State, headers storage.Headers, payloads storage.Payloads, seals storage.Seals,
+	assigner module.ChunkAssigner, verifier module.Verifier, skipSealValidity bool) *sealValidator {
+	rv := &sealValidator{
+		state:            state,
+		assigner:         assigner,
+		verifier:         verifier,
+		headers:          headers,
+		seals:            seals,
+		payloads:         payloads,
+		skipSealValidity: skipSealValidity,
+	}
+
+	return rv
+}
+
+func (s *sealValidator) verifySealSignature(aggregatedSignatures *flow.AggregatedSignature,
+	chunk *flow.Chunk, executionResultID flow.Identifier) error {
+	for i, signature := range aggregatedSignatures.VerifierSignatures {
+		signerId := aggregatedSignatures.SignerIDs[i]
+
+		nodeIdentity, err := identityForNode(s.state, chunk.BlockID, signerId)
+
+		if err != nil {
+			return err
+		}
+
+		atst := flow.Attestation{
+			BlockID:           chunk.BlockID,
+			ExecutionResultID: executionResultID,
+			ChunkIndex:        chunk.Index,
+		}
+		atstID := atst.ID()
+		valid, err := s.verifier.Verify(atstID[:], signature, nodeIdentity.StakingPubKey)
+
+		if err != nil {
+			return fmt.Errorf("failed to verify signature: %w", err)
+		}
+
+		if !valid {
+			return engine.NewInvalidInputErrorf("Invalid signature for (%x)", nodeIdentity.NodeID)
+		}
+	}
+
+	return nil
+}
+
+// Validate checks the compliance of the payload seals and returns the last
+// valid seal on the fork up to and including `candidate`. To be valid, we
+// require that seals
+// 1) form a valid chain on top of the last seal as of the parent of `candidate` and
+// 2) correspond to blocks and execution results incorporated on the current fork.
+// 3) has valid signatures for all of its chunks.
+//
+// Note that we don't explicitly check that sealed results satisfy the sub-graph
+// check. Nevertheless, correctness in this regard is guaranteed because:
+//  * We only allow seals that correspond to ExecutionReceipts that were
+//    incorporated in this fork.
+//  * We only include ExecutionReceipts whose results pass the sub-graph check
+//    (as part of ReceiptValidator).
+// => Therefore, only seals whose results pass the sub-graph check will be
+//    allowed.
+func (s *sealValidator) Validate(candidate *flow.Block) (*flow.Seal, error) {
+	// Get the latest seal in the fork that ends with the candidate's parent.
+	// The protocol state saves this information for each block that has been
+	// successfully added to the chain tree (even when the added block does not
+	// itself contain a seal). We just called `headerExtend` to check that the
+	// candidate block's header is a valid extension of the chain, which implies
+	// that the parent must already be part of the chain tree. Therefore, _not_
+	// finding the latest sealed block in the fork up to the parent constitutes
+	// a fatal internal error.
```

Comment:
⚠️ 
This comment made sense in the context of the `MutableState`, where the code was located before. However, now that we have separated the code, the comment does no longer apply, as from the perspective of the `SealValidator` there is no longer any guarantee that `MutableState.headerExtend` was called before. 

However, note that the **following is a critical requirement for the correctness of this code**:
* The SealValidator can **only process blocks which are attached to the main chain** (without any missing ancestors). 
* This is the case because:
   *  the Seal validator walks the chain backwards and requires the relevant ancestors to be known and validated
   * the `latest seal in the fork that ends with the candidate's parent` is only computed for blocks which are attached to the main chain. 

This requirement is not properly documented in the code but critical for its correctness!

 
#### Suggestion:
* Extend the GoDoc of `Validate` the method by adding the following
   ```golang
   // REQUIREMENT:
   // The SealValidator can only process blocks which are attached to the main chain 
   // (without any missing ancestors). This is the case because:
   //  * the Seal validator walks the chain backwards and requires the relevant ancestors to be known and validated
   //   * the storage.Seals only holds seals for block that are attached to the main chain. 
   ```
Please add this documentation to the _interface_ as well as the implementation.